src/Symfony/Component/HttpKernel/Bundle/SimpleBundle.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Bundle;
+
+use Symfony\Component\Config\Definition\Builder\NodeParentInterface;
+use Symfony\Component\Config\Definition\Builder\TreeBuilder;
+use Symfony\Component\Config\Definition\ConfigurationInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\HttpKernel\DependencyInjection\Extension;
+
+/**
+ * A all-in-one implementation of {@link BundleInterface},
+ * {@link ExtensionInterface} and {@link ConfigurationInterface}.
+ *
+ * @author Guilhem N. <egetick@gmail.com>
+ */
+abstract class SimpleBundle extends Bundle implements ConfigurationInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    final public function getContainerExtension()
+    {
+        return new SimpleExtension($this, function (array $config, ContainerBuilder $container) {
+            $this->buildContainer($config, $container);
+        });
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    final public function getConfigTreeBuilder()
+    {
+        $treeBuilder = new TreeBuilder();
+        $rootNode = $treeBuilder->root($this->getAlias());
+
+        $this->buildConfiguration($rootNode);
+
+        return $treeBuilder;
+    }
+
+    protected function buildConfiguration(NodeParentInterface $rootNode)
+    {
+    }
+
+    protected function buildContainer(array $config, ContainerBuilder $container)
```

Comment:
If we consider adding bundle passes to the container as part of the container building, it does build the container of course :P

The naming is kind of hard at this level, `build` is one the first methods executed (it may not be clear enough), about the new method I'd like it to be clear that it is based on configuration.
What about `configureBundle` ? or `configureContainer` ?